relay/reader/reader.go

```diff
@@ -27,6 +28,16 @@ import (
 	"github.com/pingcap/dm/pkg/log"
 )
 
+const (
+	// event timeout when trying to read events from upstream master server.
+	eventTimeout = 1 * time.Hour
```

Comment:
In order to the delay requirements of replication, let it 10 or 1minute?

Reply:
It only takes effect for getting events from _go-mysql_ (the events are already read from upstream server).
In fact, we can even use a `Cancel` rather than a `Timeout`, but use a `Timeout` we can log something to indicate still running.

the timeout between is controlled by `slaveReadTimeout` (default 1m) and `masterHeartbeatPeriod` (default 30s).

Reply:
every step will have an impact on the delay, for example, if a master-slave switch happend, there may be no data in 1h, delay would be 1h

Reply:
> every step will have an impact on the delay

agree.

> if a master-slave switch happend

An error will get with small delay

Reply:
right, set a reasonable small value and let user or dm know something ASAP

Reply:
changed to `10m` in https://github.com/pingcap/dm/pull/171/commits/0527d1660587a7f35404197d29bfc2650a9d561d.